doc/project/checklist.md

```diff
@@ -395,3 +395,91 @@ Clean up all compile-time and run-time warnings thrown by the simulator.
 ### PRE_VERIFIED_SUB_MODULES_V3
 
 Sub-modules that are pre-verified with their own testbenches have already reached V3 stage.
+
+## S1
+
+For a transition from S0 to S1, the following items are expected be completed.
+
+### DIF_EXISTS
+
+`dif_<ip>.h` and, optionally, `dif_<ip>.c` exist in `sw/device/lib/dif`.
+
+### DIF_USED_IN_TREE
+
+All existing in-tree code which uses the device, uses the device via the DIF.
+There is no remaining driver code that directly uses the device outside of DIF code.
+
+### DIF_TEST_UNIT
+
+Unit tests exist for the DIF in `sw/device/tests/dif` named `dif_<ip>_unittest.cc`.
+
+### DIF_TEST_SANITY
+
+Sanity tests exist for the DIF in `sw/device/tests/dif` named `dif_<ip>_sanitytest.c`.
+This should perform an end-to-end use of the hardware via the DIF, and should be able to be run on all OpenTitan platforms (including FPGA and simulation).
+
+## S2
+
+For a transition from S1 to S2, the following items are expected be completed.
+
+### DIF_FEATURES
+
+DIF has functions to cover all specified hardware functionality.
+
+### DIF_HW_FEATURE_COMPLETE
+
+The DIF's respective device IP is at least stage D2.
+
+### DIF_HW_PARAMS
+
+DIF uses automatically generated HW parameters and register definitions.
+
+### DIF_DOC_HW
+
+HW IP Programmer's guide references specific DIF APIs that can be used for operations.
+
+### DIF_CODE_STYLE
+
+DIF follows the DIF-specific guidelines in [`sw/device/lib/dif/README.md`]({{< relref "sw/device/lib/dif/README.md" >}}) and the OpenTitan C style guidelines.
+
+### DIF_DV_TESTS
+
+DIF is used for chip-level DV tests.
+
+### DIF_USED_TOCK
+
+DIF has initial interface for use from Tock.
+
+## S3
+
+For a transition from S2 to S3, the following items are expected be completed.
+
+### DIF_HW_DESIGN_COMPLETE
+
+The DIF's respective device IP is at least stage D3.
+
+### DIF_HW_VERIFICATION_COMPLETE
+
+The DIF's respective device IP is at least stage V3.
+
+### DIF_REVIEW_C_STABLE
+
+Fully re-review C interface and implementation, with a view to the interface not changing in future.
+
+### DIF_TEST_UNIT_COMPLETE
+
+Unit test cover (at least):
```

Comment:
Illegal Argument values are covered by "All DIF return codes". I have not yet had time to write up the guidance about DIF return codes because I'm trying to get this landed, but I have stated the guidance in a comment here: https://github.com/lowrisc/opentitan/issues/1911#issuecomment-613579290

Reply:
With regards to the phrase "Unit Test" – this is an accepted and specific software testing term.